python/src/nanoarrow/_lib.pyx

```diff
@@ -1717,6 +1717,13 @@ cdef class SchemaMetadata:
         for key, _ in self.items():
             yield key
 
+    def keys(self):
+        return list(self)
+
+    def values(self):
+        for _, value in self.items():
+            yield value
```

Comment:
Should we make `keys` and `values` return the same container type? Right now one is a list and the other is a generator. Maybe use lists for both?